### PR TITLE
Fix comment about NaN propagation

### LINKED ISSUE #7891: Incorrect comments about NaNs
**Original bug ID:** 7891
**Reporter:** sfuric
**Status:** new
**Resolution:** open
**Priority:** normal
**Severity:** minor
**Version:** 4.07.0
**Category:** documentation
**Monitored by:** @nojb @shindere @Yakobowski

## Bug description

Documentation of module Float is wrong regarding IEEE 754 evaluation rules. It says ``any arithmetic operation with nan as argument returns nan as result`` and ``Any floating-point operation with nan as argument returns nan as result`` which is wrong, as demonstrated by OCaml itself:
```
        OCaml version 4.07.0
# 1.0 ** nan;;
- : float = 1.
# nan ** 0.0;;
- : float = 1.
#
```

### PR DESCRIPTION
**Pull request by Chris00**
Fixes https://github.com/ocaml/ocaml/issues/7891

### CODE REVIEW

File: stdlib/float.mli [hunk 1]
```diff
@@ -24,8 +24,9 @@
     [neg_infinity] for [-1.0 /. 0.0], and [nan] ('not a number')
     for [0.0 /. 0.0].  These special numbers then propagate through
     floating-point computations as expected: for instance,
-    [1.0 /. infinity] is [0.0], and any arithmetic operation with [nan]
-    as argument returns [nan] as result.
+    [1.0 /. infinity] is [0.0], basic arithmetic operations
+    ([+.], [-.], [*.], [/.]) with [nan] as argument returns [nan] as
+    result,...
```
Comment (dra27) [inline stdlib/float.mli]:
```suggestion
    ([+.], [-.], [*.], [/.]) with [nan] as an argument return [nan], ...
```

File: stdlib/stdlib.mli [hunk 2]
```diff
@@ -436,8 +436,9 @@ external ( asr ) : int -> int -> int = "%asrint"
    [neg_infinity] for [-1.0 /. 0.0], and [nan] ('not a number')
    for [0.0 /. 0.0].  These special numbers then propagate through
    floating-point computations as expected: for instance,
-   [1.0 /. infinity] is [0.0], and any arithmetic operation with [nan]
-   as argument returns [nan] as result.
+    [1.0 /. infinity] is [0.0], basic arithmetic operations
+    ([+.], [-.], [*.], [/.]) with [nan] as argument returns [nan] as
+    result,...
```
Comment (dra27) [inline stdlib/stdlib.mli]:
```suggestion
    ([+.], [-.], [*.], [/.]) with [nan] as an argument return [nan], ...
```

**Review by dra27 — COMMENTED** [review]
Thanks for this! The change needs a slight tweak (return/returns disagreement) but I've suggested a further tweak to what was there before.

**Comment by dra27** [comment]
(feel to free to add a Changes entry if you'd like to)

**Review by xavierleroy — APPROVED** [review]
Looks good!  No need to delay merging.  Thanks @Chris00 for having taken care of this.

**Pull request merged**